src/lib.rs

```diff
@@ -156,6 +157,30 @@ pub unsafe extern "C" fn df_session_context_free(context: *mut SessionContext) {
     Box::from_raw(context);
 }
 
+/// Register a CSV file with the session context and return the data
+/// as a data frame.
+#[no_mangle]
+#[allow(clippy::not_unsafe_ptr_arg_deref)]
+pub extern "C" fn df_session_context_read_csv(
+    context: *mut SessionContext,
+    url: *const libc::c_char,
```

Comment:
How about adding `options` argument to avoid signature change in future?

```suggestion
    url: *const libc::c_char,
    options: *const DFCSVReadOptions,
```

In this pull request, we can always ignore it. (We can always use `CsvReadOptions::default()`.) We can use it in follow-up pull request.